internal/config/config.go

```diff
@@ -120,18 +129,14 @@ func (c Config) init(rc crt.RepoContext, creator crt.Tool) (Config, error) {
 		c.Verification.BuildRoot = verificationPaths.RemoteBuildRoot()
 	}
 
-	if c.Primary.BuildResult == "" {
-		c.Primary.BuildResult = primaryPaths.BuildResultCacheDir()
-	}
-
 	if c.Verification.BuildResult == "" {
 		c.Verification.BuildResult = verificationPaths.BuildResultCacheDir()
 	}
 
 	c.Tool = creator
 
 	if c.VerificationResult == "" {
-		c.VerificationResult = verificationPaths.VerificationResultCachePath(build.ID(c))
+		c.VerificationResult = verificationPaths.VerificationResultCachePath(ConfigIDFunc(c), c.Parameters.ZipName)
```

Comment:
Added the zip name here to make these files have the same name we're using in the action as the artifact name.